demos/libs/stop_signal.hh

```diff
@@ -0,0 +1,79 @@
+/*
+ * This file is open source software, licensed to you under the terms
+ * of the Apache License, Version 2.0 (the "License").  See the NOTICE file
+ * distributed with this work for additional information regarding copyright
+ * ownership.  You may not use this file except in compliance with the License.
+ *
+ * You may obtain a copy of the License at
+ *
+ *   http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing,
+ * software distributed under the License is distributed on an
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ * KIND, either express or implied.  See the License for the
+ * specific language governing permissions and limitations
+ * under the License.
+ */
+
+/*
+ * Copyright (C) 2020 Cloudius Systems, Ltd.
+ */
+
+#pragma once
+
+#include <seastar/core/sharded.hh>
+#include <seastar/core/reactor.hh>
+#include <seastar/core/condition-variable.hh>
+
+/// Seastar apps lib namespace
+
+namespace seastar_apps_lib {
+
+
+/// \brief Futurized SIGINT/SIGTERM signals handler class
+///
+/// Seastar-style helper class that allows easy waiting for SIGINT/SIGTERM signals
+/// from your app.
+///
+/// Example:
+/// \code
+/// #include <seastar/apps/lib/stop_signal.hh>
+/// ...
+/// int main() {
+/// ...
+/// seastar::thread th([] {
+///    seastar_apps_lib::stop_signal stop_signal;
+///    <some code>
+///    stop_signal.wait().get();  // this will wait till we receive SIGINT or SIGTERM signal
+/// });
+/// \endcode
+class stop_signal {
+    bool _caught = false;
+    seastar::condition_variable _cond;
+private:
```

Comment:
it is directly copied but from `apps/libs/stop_signal.hh` [reference](https://github.com/scylladb/seastar/blob/aff87d5bb99231a7c1836792359eb2d06517f103/apps/lib/stop_signal.hh#L31)
Should it be excluded from seastar? seems like it was added 3 years ago..

Reply:
Ah, sorry. It was copied from scylladb but the copier had permission to relicense it.

Don't create yet another copy; use the one already there.

Reply:
May be problematic, I don't know if apps/libs is accessible to demos/.

Reply:
Just double checking here before I change.
You prefer the approach of keeping the original copy in `apps/libs/stop_signal.hh` instead of moving it into `seastar/util`?
i.e include a file from the `apps` directory in the `demos` directory (I didn't take this approach initially since I didn't want to mix up the two)